### DISCUSSION TITLE
Cannot find next/google/font module in Typescript

### DISCUSSION THREAD

**Discussion by rmcsharry** [vercel/next.js#48323] [Help]
### Import Syntax Unclear For Types

The docs [here](https://nextjs.org/docs/basic-features/font-optimization) show

`import { <font-name> } from 'next/google/font';`

But [this](https://github.com/vercel/next.js/discussions/44225) discussion shows:

`import { <font-name> } from '@next/google/font';`

I have tried both, neither work. Both say module not found. The first at run-time, the 2nd at compile time

![image](https://user-images.githubusercontent.com/1156100/231669199-a1f113f7-160d-40b8-860e-5450a37cdeab.png)

It would help to at least know whether the "@" is required or not. I hope the docs are up-to-date and the discussion is not. The error that is thrown doesn't really help disambiguate these 2 syntaxes:

![Screenshot 2023-04-13 at 07 45 38](https://user-images.githubusercontent.com/1156100/231664812-d370861c-8ad8-41d0-9955-3e37cdc765bb.png)

### This is an nx mono-repo
Btw, this is in a NextJs project inside a nrwl nx mono-repo.

I tried updating the package.json to include:

    "@types/next": "13.1.1",
    
but that did not help. 

Note that the `package.json` is at the root of the monorepo (but all other deps and devDeps are there, so it must be the correct place, as sub-projects don't have their own package.json files).

### Type Declaration File
Inside the app (ie. the sub-folder) is an `index.d.ts` file where I tried adding:

`declare module 'next/font/google'`

and

`declare module '@next/font/google'`

Neither solved the problem.

### Stackoverlfow similar issue
There is [this on stackoverflow](https://stackoverflow.com/questions/74951355/where-to-place-next-font-inside-nx-workspace), but that did not help either.

### Yarn lock file
In the `yarn.lock` file is the following entry, which seems to package next inside the @nrwl namespace, so I tried '@nrwl/next/google/font' but that also did not work

```yaml
"@nrwl/next@15.8.8":
  version "15.8.8"
  resolved "https://registry.npmjs.org/@nrwl/next/-/next-15.8.8.tgz"
  integrity sha512-h7sYBf8ljyfXyoSCgiHxukB/AyZsHNL8g+vSVmr7iCkVd6IgFIa1n3myM7vaALAVrJ0EccXYdK7yuKHbVvfLnA==
  dependencies:
    "@babel/plugin-proposal-decorators" "^7.14.5"
    "@nrwl/devkit" "15.8.8"
    "@nrwl/js" "15.8.8"
    "@nrwl/linter" "15.8.8"
    "@nrwl/react" "15.8.8"
    "@nrwl/workspace" "15.8.8"
    "@svgr/webpack" "^6.1.2"
    chalk "^4.1.0"
    copy-webpack-plugin "^10.2.4"
    dotenv "~10.0.0"
    express "^4.18.1"
    fs-extra "^11.1.0"
    http-proxy-middleware "^2.0.6"
    ignore "^5.0.4"
    semver "7.3.4"
    ts-node "10.9.1"
    tsconfig-paths "^4.1.2"
    tsconfig-paths-webpack-plugin "4.0.0"
    url-loader "^4.1.1"
    webpack-merge "^5.8.0"
```

### Additional information

```js
It is a nrwl nx repo, using nrwl/next
```

I also found this similar issue, but it's not quite the same: https://github.com/vercel/next.js/discussions/44225
If I look inside my node_modules, it seems that next/font is not installed. Does nrwl not include it perhaps?

**Comment by rmcsharry** [comment]
Well the NextJs docs are not clear on this at all.

It turns out that next/font is a separate package.

So the solution looks like this, dead simple:

**package.json**

```
...
  "dependencies": {
    "@nrwl/next": "15.8.8",
    "next": "13.1.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-is": "18.2.0",
    "styled-components": "5.3.6",
    "swiper": "^9.2.0",
    "tslib": "^2.3.0",
    "@next/font": "13.1.1,"
  },
```

**_app.tsx**

```typescript
import { AppProps } from 'next/app';
import Head from 'next/head';
import 'swiper/css';
import 'swiper/css/pagination';
import './styles.css'
import { Figtree } from '@next/font/google';
// import localFont from '@next/font/local' <- if you want to use local fonts

const googleFont = Figtree({
  weight: '400',
  subsets: ['latin'],
})

function CustomApp({ Component, pageProps }: AppProps) {
  return (
    <>
      <Head>
        <title>Welcome to www.dicer.ai!</title>
      </Head>
      <main className={`app ${googleFont.className}`}>
        <Component {...pageProps} />
      </main>
    </>
  );
}

export default CustomApp;
```

**Comment by icyJoseph** [comment]
Hi,

Starting with Next 13.2, `next/font` is the name of the package, and it is built-in to the next.js core.

- https://nextjs.org/blog/next-13-2#other-improvements
- https://nextjs.org/docs/api-reference/next/font

In the docs there's a Version History details element which says:

![Screenshot 2023-04-13 at 11 35 39](https://user-images.githubusercontent.com/21013447/231718976-86a5fe2d-3557-47ad-b05f-e51e5df06ea3.png)

**Reply by g-wozniak** [reply]
Getting: `TS2307: Cannot find module next/font or its corresponding type declarations.` for Next 15.5.2 when trying to:
```
import {Nunito, Open_Sans} from 'next/font'
```

so it's still something wrong. :(

**Reply by icyJoseph** [reply]
What's your setup? every create-next-app made nowadays uses `next/font` - could we see your tsconfig file? maybe next-env.d.ts is missing?
